raitools/raitools/_managers/counterfactual_manager.py

```diff
@@ -2,33 +2,204 @@
+            self.desired_class == other_cf_config.desired_class and
+            self.permitted_range == other_cf_config.permitted_range and
+            self.features_to_vary == other_cf_config.features_to_vary
+        )
 
 
 class CounterfactualManager(BaseManager):
-    def __init__(self):
-        pass
+    def __init__(self, model, train, test, target_column, task_type):
+        """Defines the CounterfactualManager for generating counterfactuals
+           from a model.
+
+        :param model: The model to generate counterfactuals from.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param initialization_examples: A matrix of feature vector
+            examples (# examples x # features) for initializing the explainer.
+        :type initialization_examples: pandas.DataFrame
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: Task type is either 'classification/regression'
+        :type task_type: str
+        """
+        self._model = model
+        self._train = train
+        self._test = test
+        self._target_column = target_column
+        self._task_type = task_type
+        self._counterfactual_config_list = []
+
+    def _create_diceml_explainer(self, method, continuous_features):
+
+        dice_data = dice_ml.Data(dataframe=self._train,
+                                 continuous_features=continuous_features,
+                                 outcome_name=self._target_column)
+
+        model_type = 'classifier' \
+            if self._task_type == ModelTask.CLASSIFICATION else 'regressor'
+        dice_model = dice_ml.Model(model=self._model,
+                                   backend='sklearn',
+                                   model_type=model_type)
+
+        dice_explainer = Dice(dice_data, dice_model, method=method)
+
+        return dice_explainer
+
+    def _add_counterfactual_config(self, new_counterfactual_config):
+
+        if self._task_type == ModelTask.CLASSIFICATION:
+            if new_counterfactual_config.desired_class is None:
+                raise UserConfigValidationException(
+                    'The desired_class attribute should be either \'opposite\''
+                    ' or the class value for classification scenarios')
+
+            is_multiclass = len(np.unique(
+                self._train[self._target_column].values).tolist()) > 2
+            if is_multiclass and \
+                    new_counterfactual_config.desired_class == 'opposite':
+                raise UserConfigValidationException(
+                    'The desired_class attribute should not be \'opposite\''
```

Comment:
Should this specify something like opposite is not valid when the task is multiclass?

Reply:
Done